src/main/java/com/exadel/sober/controllers/AddictionController.java

```diff
@@ -0,0 +1,25 @@
+package com.exadel.sober.controllers;
+
+import com.exadel.sober.repositories.AddictionRepository;
+import com.exadel.sober.models.Addiction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@CrossOrigin
+@RequestMapping(path="/addiction")
+public class AddictionController {
+    @Autowired
+    private AddictionRepository addictionRepository;
+
+    @PostMapping(path="/add")
+    public @ResponseBody Iterable<Addiction> addNewAddiction (@RequestBody Addiction newAddiction) {
+        addictionRepository.save(newAddiction);
```

Comment:
Please find time and read about SOLID principles. You must implements single-responsibility principle, it's means one method for one task. The name of your method addNewAddiction but not addAndGetNewAddiction.